platform/ios/src/MGLMapView.mm

```diff
@@ -3246,6 +3246,12 @@ - (void)removeStyleClass:(NSString *)styleClass
     }
 
     std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:rect];
+    std::vector<MGLAnnotationTag> shapeAnnotationTags = [self shapeAnnotationTagsInRect:rect];
+    
+    if (!shapeAnnotationTags.size()) {
+        annotationTags.insert(annotationTags.end(), shapeAnnotationTags.begin(), shapeAnnotationTags.end());
```

Comment:
My point in https://github.com/mapbox/mapbox-gl-native/pull/9984#discussion_r144387593 stands, but moreover, I think this code won’t do anything: if `shapeAnnotationTags` is empty, it adds the contents of `shapeAnnotationTags` to `annotationTags`.

Reply:
@1ec5 You're right I made the corresponding changes.